app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/CameraUtils.kt

```diff
@@ -0,0 +1,209 @@
+    try {
+        val filesDir = activity.getExternalFilesDir(null)
+        val photo: Bitmap?
+        val options = BitmapFactory.Options()
+        options.inMutable = true
+        options.inDensity = 5000
+        options.inScreenDensity = 5000
+        options.inTargetDensity = 5000
+
+        photo = BitmapFactory.decodeStream(inputStream, null, options)
+
+        inputStream.close()
+
+        val originImageFile = File(filesDir, System.currentTimeMillis().toString() + "_.jpg")
+        if (originImageFile.createNewFile()) {
+            val fos = FileOutputStream(originImageFile)
+            photo!!.compress(Bitmap.CompressFormat.JPEG, 100, fos)
+            fos.close()
+        }
+        originalOutputPath = originImageFile.path
+        val logo = BitmapFactory.decodeResource(resources, R.drawable.ic_add)
+
+        //----------------------draw timestamp----------------
+        val cs = Canvas(photo)
+        val tPaint = Paint(Paint.ANTI_ALIAS_FLAG)
+
+        //get timestamp text
+        val dateTime = Date().toString()
+
+        //calculating text size
+        val textSize = photo.width / 40f
+        tPaint.isAntiAlias = true
+        tPaint.isFilterBitmap = true
+        tPaint.color = Color.WHITE
+        tPaint.style = Paint.Style.FILL
+        tPaint.textSize = textSize
+
+        val footerHeight = 1.67f * textSize
+        //draw background rect
+        tPaint.color = Color.BLACK
+        tPaint.alpha = 75
+        cs.drawRect(0f, photo.height - footerHeight, photo.width.toFloat(), photo.height.toFloat(), tPaint)
+
+        //draw texts
+        tPaint.alpha = 100
+        tPaint.textSize = textSize
+        tPaint.color = Color.WHITE
+        tPaint.style = Paint.Style.FILL
+        cs.drawText(dateTime, 20f, photo.height - textSize / 2, tPaint)
+
+        val src = Rect(0, 0, logo.width, logo.height)
+
+        val horizontalPadding = 0.2f * footerHeight
+        val logoTargetHeight = 0.9f * footerHeight
+        val scale = logoTargetHeight / logo.height
+        val logoTargetWith = scale * logo.width
+
+        val right = photo.width - horizontalPadding
+        val left = right - logoTargetWith
+        val bottom = photo.height - (footerHeight - logoTargetHeight) / 2
+        val top = bottom - logoTargetHeight
+
+        val dest = RectF(left, top, right, bottom)
+        val logoPaint = Paint(Paint.FILTER_BITMAP_FLAG or Paint.ANTI_ALIAS_FLAG or Paint.DITHER_FLAG)
+        cs.drawBitmap(logo, src, dest, logoPaint)
+        val text = "TODO application"
+        val textWith = tPaint.measureText(text)
+        cs.drawText(text, dest.left - horizontalPadding - textWith, photo.height - textSize / 2, tPaint)
+
+        //-----------------compress bitmap------------------
+        val bytes = ByteArrayOutputStream()
+        photo.compress(Bitmap.CompressFormat.JPEG, 100, bytes)
+        bytes.close()
+        photo.recycle()
+
+        //--------------------save to file----------------------
```

Comment:
You are using underscore when giving a name to file, but reading without underscore